.github/workflows/lint.yaml

```diff
@@ -1,20 +1,150 @@
+        uses: actions/checkout@v4
+      - uses: dorny/paths-filter@4512585405083f25c027a35db413c2b3b9006d50 # v2.11.1
+        id: changes
+        with:
+          base: ${{ github.event.pull_request.base.ref || github.event.merge_group.base_ref }}
+          ref: ${{ github.event.pull_request.head.ref || github.event.merge_group.head_ref }}
+          filters: |
+            go:
+              - **/*.go
+              - **/go.mod
+              - **/go.sum
+            godeps:
+              - **/go.mod
+              - **/go.sum
+            helm:
+              - examples/chart/**
+            rust:
+              - **/Cargo.lock
+              - **/Cargo.toml
+              - **/*.rs
+            proto:
+              - **/*.proto
+            web:
+              - web/**
+            terraform:
+              - '**.tf'
+              - '**.tf.json'
+              - '**.hcl'
+            sh:
+              - **/*.sh
+              - **/*.sh.tmpl
+
+  lint-terraform:
+    needs: changes
+    if: ${{ needs.changes.outputs.terraform == 'true' }}
+    uses: gravitational/shared-workflows/.github/workflows/terraform-lint.yaml@main
+    permissions:
+      actions: read
+      contents: read
+      pull-requests: write
+      security-events: write
+
+  lint-rust:
+    name: Lint (Rust)
+    needs: changes
+    if: ${{ needs.changes.outputs.rust == 'true' }}
+    runs-on: ubuntu-22.04-16core
+
+    permissions:
+      contents: read
+
+    container:
+      image: ghcr.io/gravitational/teleport-buildbox:teleport14
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        # TODO(codingllama): Consider https://github.com/actions-rs/clippy-check
+      - name: Run Rust linter
+        run: make lint-rust lint-license
+
+  lint-helm:
+    name: Lint (helm)
+    needs: changes
+    if: ${{ needs.changes.outputs.helm == 'true' }}
+    runs-on: ubuntu-22.04-16core
+    permissions:
+      contents: read
+    container:
+      image: ghcr.io/gravitational/teleport-buildbox:teleport14
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Run helm linter
+        run: make lint-helm
+
+  lint-web:
+    name: Lint (UI)
+    needs: changes
+    if: ${{ needs.changes.outputs.web == 'true' }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout OSS Teleport
+        uses: actions/checkout@v4
+      - name: Determine Toolchain Versions
+        run: |
+          echo NODE_VERSION=$(make -s -C build.assets print-node-version) >> $GITHUB_ENV
+      - name: Print versions
+        run: |
+          echo "node: ${NODE_VERSION}"
+      - name: Setup Node Toolchain
+        uses: actions/setup-node@v3
+        with:
+          node-version: ${{ env.NODE_VERSION }}
+      - name: Install Yarn dependencies
+        run: yarn --frozen-lockfile
+      - name: Run lint
+        run: yarn lint
+      - name: Run Type Check
+        run: yarn type-check
+
+  lint-sh:
```

Comment:
This new job is necessary because it used to be run indirectly along with the rust lints in `make lint-no-actions`.